saleor/payment/gateway.py

```diff
@@ -128,6 +154,10 @@ def capture(
     )
     if response and response.payment_method_info:
         update_payment_method_details(payment, response)
+    if response and response.transaction_already_processed:
+        transaction = get_already_processed_transaction(payment, response)
+        if transaction:
+            return transaction
```

Comment:
This code is duplicated in many places. Maby we should extract it to some utils?